### PR TITLE
Replace publicKeys/privateKeys with encryptionKeys, decryptionKeys, signingKeys, and verificationKeys

### PR DESCRIPTION
**Pull request by alichry**
This PR aims to fulfill https://github.com/openpgpjs/openpgpjs/pull/1205#issuecomment-772888464

Notes:

- Some error messages were modified:
  - `Private key is not decrypted.` --> `Decryption key is not decrypted.`
  -  ` detached option has been removed from openpgp.encrypt. Separately call openpgp.sign instead. Don't forget to remove privateKeys option as well.` --> `detached option has been removed from openpgp.encrypt. Separately call openpgp.sign instead. Don't forget to remove signingKeys option as well.`
  -  `Public keys are required to verify message signatures` --> `Verification keys are required to verify message signatures`
- Should the `keys` parameters be renamed in `Message.prototype.createVerificationObjects?`
`Message.prototype.encrypt`, `Message.prototype.verify(Detached)?`,
and `Message.prototype.encryptSessionKeys` ?
- JSDoc for `message.js` might describe the `encryptionKeys` parameter as a list of **public** keys. Similarly for `decryptionKeys` and the like. Just to hint that the encryption key is often a public key. Should this be changed?

Checklist:

- [x] openpgp.encrypt
- [x] openpgp.decrypt
- [x] openpgp.sign
- [x] openpgp.verify
- [x] openpgp.generateSessionKey
- [x] openpgp.encryptSessionKey
- [x] openpgp.decryptSessionKey
- [x] openpgp.signDetached
- [x] Message.prototype.encrypt
- [x] Message.prototype.decrypt
- [x] Message.prototype.sign
- [x] Message.prototype.signDetached
- [x] Message.prototype.verify
- [x] Message.prototype.createSignaturePackets
- [x] Message.prototype.generateSessionKey
- [x] Message.prototype.encryptSessionKey
- [x] Message.prototype.decryptSessionKey
- [x] Fix test cases

### CODE REVIEW

**Comment by larabr** [comment]
Thanks for implementing this.
I would only change the params of the top-level `openpgp.*` functions, since the other methods are somewhat meant to be internal, and at a low-level I find `privateKeys` and `publicKeys` to be clearer -- otherwise there can be some confusion between `passphrase`/`decryption/encryptionSessionKey` and `encryption/decryptionKey`. But wait for @twiss 's feedback on this :)

About the JS doc, I wouldn't worry too much in this PR since we are going to implement #1293 soon.

File: src/message.js [hunk 1]
```diff
@@ -359,22 +359,22 @@ export class Message {
    * @param {Uint8Array} sessionKey - session key for encryption
    * @param {String} algorithm - session key algorithm
    * @param {String} [aeadAlgorithm] - AEAD algorithm, e.g. 'eax' or 'ocb'
-   * @param {Array<Key>} [publicKeys] - Public key(s) for message encryption
+   * @param {Array<Key>} [encryptionKeys] - Encryption key(s) for message encryption
```
Comment (twiss) [inline src/message.js:362]:
```suggestion
   * @param {Array<Key>} [encryptionKeys] - Public key(s) for message encryption
```

**Review by twiss — COMMENTED** [review]
Hey @alichry, thanks a lot for the PR!

Tbh, I'm fine with changing the param names also in the low-level functions. For example, in `encryptSessionKey(sessionKey, ..., encryptionKeys, ...)` it arguably actually makes it more obvious that it's encrypting the `sessionKey` with the `encryptionKeys`.

> * `detached option has been removed from openpgp.encrypt. Separately call openpgp.sign instead. Don't forget to remove privateKeys option as well.` --> `detached option has been removed from openpgp.encrypt. Separately call openpgp.sign instead. Don't forget to remove signingKeys option as well.`

This one should actually keep saying `privateKeys`, or maybe `privateKeys/signingKeys`, the reason is that it's referring to how `openpgp.sign` used to be in the past, but that they should stop doing that.

> * Should the `keys` parameters be renamed in `Message.prototype.createVerificationObjects?`
>   `Message.prototype.encrypt`, `Message.prototype.verify(Detached)?`,
>   and `Message.prototype.encryptSessionKeys` ?

If you want, go for it :+1:

> JSDoc for message.js might describe the encryptionKeys parameter as a list of public keys. Similarly for decryptionKeys and the like. Just to hint that the encryption key is often a public key. Should this be changed?

No, this can be kept indeed. In the ideal case it would mention both, as in the suggestion below, but as Lara said, don't worry about it too much for now, this will become more clear once we have dedicated types for these.

**Comment by alichry** [comment]
Thank you both for the review ^-^ I will push a some commits soon

**Comment by alichry** [comment]
I believe the test cases should pass now :)

File: src/message.js [hunk 2]
```diff
@@ -543,7 +543,7 @@ export class Message {
    * }>>} List of signer's keyID and validity of signatures.
    * @async
    */
-  async verify(keys, date = new Date(), config = defaultConfig) {
+  async verify(encryptionKeys, date = new Date(), config = defaultConfig) {
```
[the remark this answers is not in the document]
Comment (alichry) [reply src/message.js]: Done! Thanks a lot :)

File: src/message.js [hunk 3]
```diff
@@ -543,7 +543,7 @@ export class Message {
    * }>>} List of signer's keyID and validity of signatures.
    * @async
    */
-  async verify(keys, date = new Date(), config = defaultConfig) {
+  async verify(signingKeys, date = new Date(), config = defaultConfig) {
```
[the remark this answers is not in the document]
Comment (alichry) [reply src/message.js:546]: :0, Let me fix that shortly :)!

File: openpgp.d.ts [hunk 4]
```diff
@@ -240,18 +240,18 @@ export class Message<T extends MaybeStream<Data>> {
...
   /** Unwrap compressed message
    */
   public unwrapCompressed(): Message<T>;
 
   /** Verify message signatures
-      @param keys array of keys to verify signatures
+      @param signingKeys array of public keys to verify signatures
   */
-  public verify(keys: Key[], date?: Date, config?: Config): Promise<VerificationResult[]>;
+  public verify(signingKeys: Key[], date?: Date, config?: Config): Promise<VerificationResult[]>;
```
[the remark this answers is not in the document]
Comment (alichry) [reply openpgp.d.ts]: There you go ^-^ ✌️

**Review by twiss — APPROVED** [review]

File: openpgp.d.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (twiss) [reply openpgp.d.ts]: (Thanks again! :blush:)

**Comment by alichry** [comment]
🥳🥳 thank you for your thorough review and merging it!! ✌️

**Pull request merged**